pkg/controller/migmigration/hooks.go

```diff
@@ -0,0 +1,203 @@
+package migmigration
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	migapi "github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (t *Task) runHooks() (bool, error) {
+	hook := migapi.MigPlanHook{}
+	var client k8sclient.Client
+	var err error
+
+	for _, h := range t.PlanResources.MigPlan.Spec.Hooks {
+		if (h.Phase + "Hooks") == t.Phase {
```

Comment:
Correct.  I could create a constant, but wasn't sure it was worth it for one location. I could also do a contains or something if it makes sense.

Reply:
Currently, the `Phase` is internal to the migration controller.  The `Phase` is deprecated by kubernetes but used here because: 1) It's too practical to resist; 2) It's internal to the controller and only exposed to the user as informational only.  This encapsulation ensures that phases can be renamed, reordered and re-implemented with side effects.  The correlation to `MigPlanHook` violates this.  This is why I suggested that plan spec include 4 optional hook references instead of correlating by phase.

About the _"Hook"_ suffix:

I assume the goal for making the `MigPlanHook.Phase` different than the itinerary phase is so the user does not need to include the _"Hooks"_ suffix.  I'm not convinced this is valuable enough to make them different and requiring the correlation.  Also, it may be confusing to the user since it won't match the reported status (phase) of the migration.

A few suggested options should we continue with using the phase to correlate hooks.

Perhaps passing the `MigPlanHook.Phase` to `runHooks()` to use for the match criteria instead of relying on a naming convention would be less _magical_?  

Calling in the task.Run():
```
case PreBackupHooks:
    t.runHooks("PreBackup")
```

Another consideration would be to drop the _"Hooks_" suffix from the itinerary phases.

Reply:
@jortel @djwhatle I think my last commit accomplished what was desired from our conversation yesterday. If not please let me know.